pkg/hnsw/hnsw.go

```diff
@@ -109,9 +109,12 @@ func (h *Hnsw) searchLayer(q Vector, entryNode *Node, ef int, layerId int) (*Bas
 			break
 		}
 
+		fmt.Println("we should be checking")
+
 		friends := h.Nodes[closestCandidate.id].GetFriendsAtLevel(layerId)
```

Comment:
specifically here, when we try to `GetFriendsAtLevel`, the entry node has no friends, as the graph has been freshly created